module/VuFind/src/VuFind/Http/PhpEnvironment/RemoteAddressFactory.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * RemoteAddress utility factory. This uses the core Zend RemoteAddress but
+ * configures it according to VuFind settings.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Http\PhpEnvironment;
+
+use Interop\Container\ContainerInterface;
+use Zend\ServiceManager\Factory\FactoryInterface;
+
+/**
+ * RemoteAddress utility factory.
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class RemoteAddressFactory implements FactoryInterface
+{
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException if any other error occurs
+     */
+    public function __invoke(ContainerInterface $container, $requestedName,
+        array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options sent to factory.');
+        }
+        $cfg = $container->get(\VuFind\Config\PluginManager::class)->get('config');
+        $object = new $requestedName();
+        if ($cfg->Site->reverse_proxy ?? false) {
```

Comment:
This appears to be an undocumented setting. Can you add some notes to config.ini? Is this something that actually belongs here, or should this be committed separately?

Reply:
Not something I came up with, but I'm pretty proud of having taken it into account. ;) See:

https://github.com/vufind-org/vufind/blob/master/module/VuFind/src/VuFind/View/Helper/Root/ServerUrlFactory.php#L67
https://github.com/vufind-org/vufind/blob/master/config/vufind/config.ini#L29

Reply:
I don't know what I was looking at, but clearly it was the wrong thing. This was added in #1429, and as you say, it's already documented. Sorry about the confusion (and good work accounting for it). :-)